aks-node-controller/app.go

```diff
@@ -95,6 +102,48 @@ func (a *App) Provision(ctx context.Context, flags ProvisionFlags) error {
 	return err
 }
 
+func (a *App) ProvisionWait(ctx context.Context) (string, error) {
```

Comment:
Would be nice to have a test watching for a file in tmp folder

Reply:
Going to add this in a follow-up PR, and maybe implement the IO interface as you suggested for mocking the file operations.